src/odemis/gui/comp/overlay/world.py

```diff
@@ -2968,259 +2944,107 @@ def draw(self, ctx, shift=(0, 0), scale=1.0):
-        self.drag_v_end_y = None  # the current y coordinate of the mouse position
+        # The X position is fixed: from the very left till the very right
+        # The Y position depends on the wavelength: pos (from the center, in m) =  a + b * wl
+        a, b = line
+        p_start_y =  a + b * wl_start
+        p_end_y = a + b * wl_end
+        self.p_start_pos = self.det_edges[0], p_start_y
+        self.p_end_pos = self.det_edges[1], p_end_y
 
-        self._edges = {}
-        self.mode = NONE_LINE_MODE
+        # Force an update of the view
+        self._phys_to_view()
+        # self.cnvs.request_drawing_update()
+        wx.CallAfter(self.cnvs.request_drawing_update)
 
     def _view_to_phys(self):
-        """ Update the physical position to reflect the view position """
-        if self.v_start_pos and self.v_end_pos:
+        super()._view_to_phys()
 
-            offset = self.cnvs.get_half_buffer_size()
+        # TODO: still can get confused when the wl is changed, but the canvas hasn't been redrawn
+        # (eg, because image is paused, or long exp time).
+        # => listen to the spectrograph.position in the EKOverlay and force reading redrawing a little later?)
 
-            self.p_start_pos = self.cnvs.view_to_phys(self.v_start_pos, offset)
-            self.p_end_pos = self.cnvs.view_to_phys(self.v_end_pos, offset)
-
-            self._calc_edges()
-
-    def start_dragging(self, drag, vpos):
-        """
-        The user can start moving line or editing its endpoints only along the y axis.
-        """
-        super(EkLine, self).start_dragging(drag, vpos)
-
-        self.drag_v_start_y = self.drag_v_end_y = vpos[1]
-
-        if drag == gui.HOVER_START:
-            self.mode = EDIT_START_EDGE_MODE
-        elif drag == gui.HOVER_END:
-            self.mode = EDIT_END_EDGE_MODE
-        elif drag == gui.HOVER_LINE:
-            self.mode = MOVE_LINE_MODE
-        else:
-            raise ValueError("No valid hover mode")
-
-    def on_motion(self, vpos, ctrl_down=None):
-        """
-        Given that the left mouse button is already pressed down and the mouse cursor is over the line,
-        the user can move line along y axis until the left button is released.
-        """
-
-        self.drag_v_end_y = vpos[1]
-        super(EkLine, self).on_motion(vpos, ctrl_down)
+        # Update the VA
+        try:
+            wl_start = self._wl_list[0]
+            wl_end = self._wl_list[-1]
+        except (IndexError, TypeError):
+            return
 
-        if self.mode in (EDIT_START_EDGE_MODE, EDIT_END_EDGE_MODE):
-            self._update_editing()
-        else:  # self.mode == MOVE_LINE_MODE:
-            self._update_moving()
-        self.cnvs.Refresh()
+        # Reverse equation of pos =  a + b * wl
+        b = (self.p_end_pos[1] - self.p_start_pos[1]) / (wl_end - wl_start)
+        logging.debug("%s mm / %s nm", 1000 * (self.p_end_pos[1] - self.p_start_pos[1]), (wl_end - wl_start) * 1e9)
```

Comment:
Can you put the multiplication with 1000 also behind the values, to make it easier to read:
```
logging.debug("%s mm / %s nm", (self.p_end_pos[1] - self.p_start_pos[1]) * 1000, (wl_end - wl_start) * 1e9)
```